src/pyproject_hooks/_in_process/_in_process.py

```diff
@@ -106,6 +106,13 @@ def find_spec(self, fullname, _path, _target=None):
 
         return spec
 
+    def find_distributions(self, context=None):
+        # Delayed import: Python 3.7 does not contain importlib.metadata
+        from importlib.metadata import DistributionFinder, MetadataPathFinder
```

Comment:
The `importlib_metadata` provides these interfaces going back to Python 2.7 (and maybe 2.6). If this project wishes to support Python 3.7, it should consider using `importlib_metadata`.

Reply:
Yes, that is something on the back of my mind: what would be the best approach here to avoid errors in 3.7 if `importlib_metadata` is being used, but without adding a new dependency on this project.

Initially I thought to minimise the complexity, but it is true that the initial implementation is error prone.

I added https://github.com/pypa/pyproject-hooks/pull/195/commits/455b77f9a89f3bfded5925ccc15e17d20c9b99cd to account for that. Not sure if the best is to try first `importlib_metadata` and fallback to `importlib.metadata` or the other way around (please let me know if the opposite is better).

@takluyver I understand that it is the approach in this module to keep supporting 3.7, right?

Reply:
Agreed that adding `importlib_metadata` adds an additional dependency. It also provides forward-compatibility for features introduced in later Pythons. I do see that in 4.0 (Python 3.10), there was improved alignment with naming standards, so that could affect discovery. For setuptools, I don't think that's a problem.